processor/otel/metrics.go

```diff
@@ -89,6 +90,172 @@ func (c *Consumer) convertResourceMetrics(resourceMetrics pdata.ResourceMetrics,
 	}
 }
 
+// Otel specification : https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/system-metrics.md
+type apmMetricsBuilder struct {
+	// Host CPU metrics
+	cpuCount                 int // from system.cpu.utilization's cpu attribute
+	nonIdleCPUUtilizationSum apmMetricValue
+
+	// Host memory metrics
+	freeMemoryBytes apmMetricValue
+	usedMemoryBytes apmMetricValue
+
+	// JVM metrics
+	jvmGCTime  map[string]apmMetricValue
+	jvmGCCount map[string]apmMetricValue
+	jvmMemory  map[jvmMemoryKey]apmMetricValue
+}
+
+func NewApmMetricsBuilder() *apmMetricsBuilder {
+	var b apmMetricsBuilder
+	b.jvmGCTime = make(map[string]apmMetricValue)
+	b.jvmGCCount = make(map[string]apmMetricValue)
+	b.jvmMemory = make(map[jvmMemoryKey]apmMetricValue)
+	return &b
+}
+
+type apmMetricValue struct {
+	timestamp time.Time
+	value     float64
+}
+
+type jvmMemoryKey struct {
+	area  string
+	type_ string
+	pool  string // will be "" for non-pool specific memory metrics
+}
+
+// accumulate processes m, translating to and accumulating equivalent Elastic APM metrics in b.
+func (b *apmMetricsBuilder) accumulate(m pdata.Metric) {
+
+	switch m.DataType() {
+	case pdata.MetricDataTypeSum:
+		dpsCounter := m.Sum().DataPoints()
+		for i := 0; i < dpsCounter.Len(); i++ {
+			dp := dpsCounter.At(i)
+			if sample, ok := numberSample(dp, model.MetricTypeCounter); ok {
+				switch m.Name() {
+				case "system.memory.usage":
+					if memoryState, exists := dp.Attributes().Get("state"); exists {
+						switch memoryState.StringVal() {
+						case "used":
+							b.usedMemoryBytes = apmMetricValue{dp.Timestamp().AsTime(), sample.Value}
+						case "free":
+							b.freeMemoryBytes = apmMetricValue{dp.Timestamp().AsTime(), sample.Value}
+						}
+					}
+				case "runtime.jvm.gc.collection", "runtime.jvm.gc.time":
+					if gcName, exists := dp.Attributes().Get("gc"); exists {
+						b.jvmGCTime[gcName.StringVal()] = apmMetricValue{dp.Timestamp().AsTime(), sample.Value}
+					}
+				case "runtime.jvm.gc.count":
+					if gcName, exists := dp.Attributes().Get("gc"); exists {
+						b.jvmGCCount[gcName.StringVal()] = apmMetricValue{dp.Timestamp().AsTime(), sample.Value}
+					}
+				}
+			}
+		}
+	case pdata.MetricDataTypeGauge:
+		// Gauge metrics accumulation
+		dpsGauge := m.Gauge().DataPoints()
+		for i := 0; i < dpsGauge.Len(); i++ {
+			dp := dpsGauge.At(i)
+			if sample, ok := numberSample(dp, model.MetricTypeGauge); ok {
+				switch m.Name() {
+				case "system.cpu.utilization":
+					if cpuState, exists := dp.Attributes().Get("state"); exists {
+						if cpuState.StringVal() != "idle" {
+							if sample.Value > 1 {
+								b.nonIdleCPUUtilizationSum.value += 1
+							} else {
+								b.nonIdleCPUUtilizationSum.value += sample.Value
+							}
```

Comment:
This check was introduced due to the `host-metrics` [package](https://github.com/open-telemetry/opentelemetry-js-contrib/tree/main/packages/opentelemetry-host-metrics) of Opentelemetry-Js being implemented in a way that results in utilisation statistics being sometimes larger than 1. This check is used as a threshold.

We should translate the specification "as-is", so I believe that we should remove this check. Do you have an opinion on this ?

NB : For your information, the cause of the issue is here : 
https://github.com/open-telemetry/opentelemetry-js-contrib/blob/0d90e9d1f1cb5e09ce8f50c17bc866d02df34512/packages/opentelemetry-host-metrics/src/stats/common.ts#L35
For example, `userP`, which is the CPU utilisation in user mode, is computed as the ratio between `os.cpus.times.user` and `timeElapsed`. The first one is the time spent by the CPU on user processes since boot, while the second is the time allocated to the instrumented process since it started. If the host has been up and running for a long time, this ratio can go well beyond 1. I actually have to open an issue on this.

Reply:
Thanks for the explanation. I would prefer that we don't work around bugs in instrumentation. Let's stick to the spec like you say, and +1 on opening an issue on opentelemetry-js-contrib.

Reply:
Ok, thanks for the insights. I have removed the check in 8d501af.